### PR TITLE
Convert to topkg

### LINKED ISSUE #13: OSX build is broken, trying to link with -lrt
#12 applies `-lrt` to all builds in unix, causing errors in OSX

I've converted to this repo to `topkg`, which made fixing this issue easier (I couldn't figure out how to apply the fix only to Linux using Oasis). A PR for that is on it's way shortly.

```
#=== ERROR while installing mirage-clock-unix.dev~mirage ======================#
# opam-version 1.2.2
# os           darwin
# command      make unix-build
# path         /Users/mgray/.opam/horseos/build/mirage-clock-unix.dev~mirage
# compiler     system (4.03.0)
# exit-code    2
# env-file     /Users/mgray/.opam/horseos/build/mirage-clock-unix.dev~mirage/mirage-clock-unix-92783-c743ac.env
# stdout-file  /Users/mgray/.opam/horseos/build/mirage-clock-unix.dev~mirage/mirage-clock-unix-92783-c743ac.out
# stderr-file  /Users/mgray/.opam/horseos/build/mirage-clock-unix.dev~mirage/mirage-clock-unix-92783-c743ac.err
### stdout ###
# pkg_lwt_unix: ........................................ /Users/mgray/.opam/horseos/lib/lwt
# [...]
# ocaml setup.ml -build
# ocamlopt.opt unix.cmxa -I /Users/mgray/.opam/horseos/lib/ocamlbuild /Users/mgray/.opam/horseos/lib/ocamlbuild/ocamlbuildlib.cmxa myocamlbuild.ml /Users/mgray/.opam/horseos/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# /Users/mgray/.opam/horseos/bin/ocamlfind ocamlc -g -ccopt -O2 -ccopt -g -ccopt -Wall -c unix/clock_stubs.c
# mv clock_stubs.o unix/clock_stubs.o
# /Users/mgray/.opam/horseos/bin/ocamlfind ocamlmklib -o unix/mirage-clock-unix_stubs -lrt unix/clock_stubs.o
# + /Users/mgray/.opam/horseos/bin/ocamlfind ocamlmklib -o unix/mirage-clock-unix_stubs -lrt unix/clock_stubs.o
# ld: library not found for -lrt
# clang: error: linker command failed with exit code 1 (use -v to see invocation)
# Command exited with code 2.
### stderr ###
# E: Failure("Command ''/Users/mgray/.opam/horseos/bin/ocamlbuild' unix/libmirage-clock-unix_stubs.a unix/dllmirage-clock-unix_stubs.so unix/mirage-clock-unix.cma unix/mirage-clock-unix.cmxa unix/mirage-clock-unix.a unix/mirage-clock-unix.cmxs lib_test/portable.native -tag debug' terminated with error code 10")
# make: *** [unix-build] Error 1
```

### PR DESCRIPTION
**Pull request by mattgray**
This fixes #13, and I included #6 as topkg requires a license file.

Tested with Dockerfile like:

```
FROM ocaml/opam:ubuntu-12.04_ocaml-4.03.0

RUN opam repo add mirage-dev git://github.com/mirage/mirage-dev
RUN opam depext -i mirage
RUN opam pin add mirage-clock-unix git://github.com/mattgray/mirage-clock#topkg
RUN git clone -b mirage-dev http://github.com/mirage/mirage-skeleton
WORKDIR /home/opam/mirage-skeleton/console
RUN opam config exec -- mirage configure -t unix
RUN opam config exec make
```

for unix and xen.

I think I've used topkg correctly, @dbuenzli might want to look over this, the `pkg.ml` contains 2 opam packages, tests, and use of C stubs

### CODE REVIEW

File: mirage-clock-unix.opam [hunk 1]
```diff
@@ -3,18 +3,22 @@ authors: "The MirageOS team"
...
-  "ocamlfind"
+  "ocamlfind" {build}
+  "ocamlbuild" {build}
+  "topkg" {build}
   "mirage-types" {>= "0.3.0"}
   "mirage-types-lwt"
   "lwt"
-  "cstruct" {>= "2.0.0"}
+]
+build: [ "ocaml" "pkg/pkg.ml" "build" "--pkg-name" name "--pinned" pinned ]
```
Comment (dbuenzli) [inline mirage-clock-unix.opam]: If this has to go through the current Ocaml OPAM repository, CI will scream on `pinned` replace with `"%{pinned}%"`

File: mirage-clock-xen.opam [hunk 2]
```diff
@@ -3,17 +3,21 @@ authors: "The MirageOS team"
...
+  "ocamlbuild" {build}
+  "topkg" {build}
   "mirage-types" {>= "0.3.0"}
   "mirage-types-lwt"
   "lwt"
 ]
+build: [
+  "ocaml" "pkg/pkg.ml" "build"
+            "--pkg-name" name
+            "--pinned" pinned ]
```
Comment (dbuenzli) [inline mirage-clock-xen.opam]: Here aswell.

File: unix/META [hunk 3]
```diff
@@ -1,12 +1,8 @@
-# OASIS_START
-# DO NOT EDIT (digest: 010e47cbabd0a2dbc2c1b40fb4ac99cd)
 version = "1.1.0"
```
Comment (dbuenzli) [inline unix/META]: This should be `%%VERSION%%` rather than `1.1.0`

File: xen/META [hunk 4]
```diff
@@ -1,12 +1,8 @@
-# OASIS_START
-# DO NOT EDIT (digest: 3e242c845e7e58d933f8a63847acd1e7)
 version = "1.1.0"
```
Comment (dbuenzli) [inline xen/META]: Again.

**Comment by mattgray** [comment]
this is truly fixed now, previously the -lrt flag wasn't being passed to the test build in Linux

**Comment by yomimono** [comment]
I'm having Travis build mirage-skeleton against this and I'll merge once it passes.

**Comment by yomimono** [comment]
Looks like all's well: https://travis-ci.org/yomimono/mirage-skeleton/jobs/159593898

**Pull request merged**